### PR TITLE
Document the GlobalConfig resource as a TransportServer pre-requisite

### PR DESCRIPTION
**Pull request by soneillf5**
### Proposed changes
Add additional documentation for GlobalConfig as a TransportServer pre-requisite.

### Checklist
Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [CONTRIBUTING](https://github.com/nginxinc/nginx-ingress-operator/blob/master/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that all unit tests pass after adding my changes
- [x] I have updated necessary documentation
- [x] I have rebased my branch onto master
- [x] I will ensure my PR is targeting the master branch and pulling from my branch from my own fork

### CODE REVIEW

**Review by ciarams87 — COMMENTED** [review]
Looks great, but should we add a link to the TransportServers header to the base README as well in the Getting started section -> https://github.com/nginxinc/nginx-ingress-operator#getting-started? The normal flow of the docs is: "1. Deploy the operator. 2. Read these examples to deploy the IC" so it might be too late for this to be the first reference.

**Review by vepatel — APPROVED** [review]

File: examples/deployment-oss-min/README.md [hunk 1]
```diff
@@ -52,4 +53,21 @@ For more information about how to configure the NGINX Ingress Controller, check
...
     kubectl delete namespace my-nginx-ingress
-    ```
\ No newline at end of file
+    ```
+   
+## TransportServers
+
+TransportServers provide support for TCP/UDP but are in active development and provided as a preview feature. 
+A GlobalConfig resource is used to specify the TCP/UDP listeners and is required by TransportServers. 
+To use TransportServers, you must create a GlobalConfig resource *before* starting the Operator.
```
Comment (vepatel) [inline examples/deployment-oss-min/README.md:62]: should we specify that it has to be created after namespace(s)?

Comment (soneillf5) [reply examples/deployment-oss-min/README.md:62]: updated

**Review by ciarams87 — APPROVED** [review]

File: examples/deployment-oss-min/README.md [hunk 2]
```diff
@@ -52,4 +53,29 @@ For more information about how to configure the NGINX Ingress Controller, check
...
+Step 2. global configuration <--- in this order
+Step 3. ingress controller
+...
+```
+
+```
+kubectl apply -f global-configuration.yaml
+```
+
+Then update the NginxIngressController to use the global configuration by adding the following config to `nginx-ingress-controller.yaml`
```
Comment (pleshakov) [inline examples/deployment-oss-min/README.md:76]:
```suggestion
Then update the NginxIngressController to use the GlobalConfiguration by adding the following config to `nginx-ingress-controller.yaml`
```

**Review by pleshakov — COMMENTED** [review]
Hi @soneillf5 

Looks good

I added a few clarifications and some small improvements. 

Also, could you update the table here ? We can specify the requirement for the field existence in the doc of that field
image

For example, something like this: The Ingress Controller will fail to start if the GlobalConfiguration resource doesn't exist.

Consider a similar doc for the wildcard secret 
image

In case of a doc update for a field, it is also necessary to update the go docs https://github.com/nginxinc/nginx-ingress-operator/blob/master/pkg/apis/k8s/v1alpha1/nginxingresscontroller_types.go and regenerate CRDs - for consistency.

**Pull request merged**